llvm/lib/CodeGen/SelectionDAG/TargetLowering.cpp

```diff
@@ -8639,6 +8639,24 @@ SDValue TargetLowering::expandCTPOP(SDNode *Node, SelectionDAG &DAG) const {
   if (VT.isVector() && !canExpandVectorCTPOP(*this, VT))
     return SDValue();
 
+  // i8 CTPOP - with efficient i32 MUL, then attempt multiply-mask-multiply.
+  if (VT == MVT::i8 && shouldAllowMultiplyInBitCounts(MVT::i8, MVT::i32) &&
```

Comment:
Do we really need yet another extremely specific target hook? I would expect the isOperationLegal checks be the base implementation if so

Reply:
So far only x86 (and maybe arm) have benefited from this

Reply:
Would checking the immediate cost be able to distinguish targets where this is profitable?